libraries/c_sdk/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -606,8 +613,19 @@ static void _destroyMqttConnection( _mqttConnection_t * pMqttConnection )
     {
         IotMutex_Delete( &( connToContext[ contextIndex ].contextMutex ) );
         IotMutex_Delete( &( connToContext[ contextIndex ].subscriptionMutex ) );
-        /* Free the MQTT Context for the MQTT connection to be destroyed. */
-        _IotMqtt_removeContext( pMqttConnection );
+        /* Lock mutex before updating the #connToContext array. */
+        connContextMutexStatus = IotMutex_TakeRecursive( &connContextMutex );
+
+        if( connContextMutexStatus == true )
+        {
+            /* Free the MQTT Context for the MQTT connection to be destroyed. */
+            _IotMqtt_removeContext( pMqttConnection );
+
+            /* Lock mutex before updating the #connToContext array. */
+            connContextMutexStatus = IotMutex_TakeRecursive( &connContextMutex );
```

Comment:
Do you need to take it again?  Should this be a give?

Reply:
No. Thanks for catching it. Will fix it.